app/src/main/java/org/mozilla/fenix/onboarding/view/OnboardingViewModel.kt

```diff
@@ -0,0 +1,158 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.onboarding.view
+
+import androidx.annotation.DrawableRes
+import androidx.annotation.StringRes
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import kotlinx.coroutines.flow.MutableSharedFlow
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.SharedFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.launch
+import mozilla.telemetry.glean.private.NoExtras
+import org.mozilla.fenix.GleanMetrics.Onboarding
+import org.mozilla.fenix.R
+
+/**
+ * ViewModel holding the screen state for [OnboardingScreen].
+ */
+class OnboardingViewModel : ViewModel() {
```

Comment:
We have moved away from using `ViewModel` in favour of [Store](https://searchfox.org/mozilla-mobile/source/firefox-android/android-components/components/lib/state/src/main/java/mozilla/components/lib/state/Store.kt), which follows more closely with [Redux](https://redux.js.org/). 

Some examples:
https://searchfox.org/mozilla-mobile/source/fenix/app/src/main/java/org/mozilla/fenix/tabstray/TabsTrayStore.kt
https://searchfox.org/mozilla-mobile/search?q=Store&path=&case=false&regexp=false

Reply:
Good call! this logic in the VM can be moved to the Store and reducer can handle the actions. Yes, the intention was to move the state and logic outside the composables and even some of the navigation events that depend on the state. I'm going to explore this.

Reply:
The big advantage of a `ViewModel` is that it survives configuration changes. Which in the current architecture and with the app handling screen rotations (with the UI not being destroyed) doesn't warrant it's a bit more complex machinery.
With the reducers not having access to other dependencies and them being `pure` functions the domain logic would usually stay in `Controller`s (`Interactor`s are also being phased out with them having little to no separate responsability) - https://github.com/mozilla-mobile/fenix/blob/main/docs/architecture-overview.md.